clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/spark/input/hdfs/VeniceHdfsInputPartitionReader.java

```diff
@@ -46,25 +35,15 @@ protected VeniceRecordIterator createRecordIterator(VeniceProperties jobConfig,
       throw new VeniceException("Unable to get a FileSystem", e);
     }
 
-    String keyFieldStr = jobConfig.getString(KEY_FIELD_PROP, DEFAULT_KEY_FIELD_PROP);
-    String valueFieldStr = jobConfig.getString(VALUE_FIELD_PROP, DEFAULT_VALUE_FIELD_PROP);
-    ETLValueSchemaTransformation etlValueSchemaTransformation = ETLValueSchemaTransformation
-        .valueOf(jobConfig.getString(ETL_VALUE_SCHEMA_TRANSFORMATION, ETLValueSchemaTransformation.NONE.name()));
     boolean vsonPush = jobConfig.getBoolean(VSON_PUSH, false);
     Path filePath = inputPartition.getFilePath();
 
     if (vsonPush) {
-      String fileKeySchema = jobConfig.getString(FILE_KEY_SCHEMA);
-      String fileValueSchema = jobConfig.getString(FILE_VALUE_SCHEMA);
-      VeniceVsonRecordReader recordReader =
-          new VeniceVsonRecordReader(fileKeySchema, fileValueSchema, keyFieldStr, valueFieldStr);
+      VeniceVsonRecordReader recordReader = new VeniceVsonRecordReader(jobConfig);
       return new VeniceVsonFileIterator(fs, filePath, recordReader);
     }
 
-    // Input data is not in Vson format. It must be Avro.
-    Schema fileSchema = AvroCompatibilityHelper.parse(jobConfig.getString(SCHEMA_STRING_PROP));
```

Comment:
It seems like the fromProps implementation is doing slightly more strict parsing check, which I think it is fine:`    Schema dataSchema = AvroSchemaParseUtils.parseSchemaFromJSON(
        props.getString(SCHEMA_STRING_PROP),
        props.getBoolean(EXTENDED_SCHEMA_VALIDITY_CHECK_ENABLED, DEFAULT_EXTENDED_SCHEMA_VALIDITY_CHECK_ENABLED));
`
Orginal code is doing loose check.
Personally I think it is OK because it is data schema. Just poin this out, If you think it is fine as well, just close this comment.

Reply:
I think it is fine because the VPJ driver would have already done this same schema check. Also, this is the same code that MR mode is running: https://github.com/linkedin/venice/blob/main/clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/mapreduce/datawriter/map/VeniceAvroMapper.java#L14